cmake/onnxruntime_mlas.cmake

```diff
@@ -251,6 +265,49 @@ function(setup_mlas_source_for_windows)
   endif()
 endfunction()
 
+function(setup_kleidiai)
+  set(onnxruntime_USE_KLEIDIAI ${ARM64} CACHE BOOL "")
+  target_compile_definitions(onnxruntime_mlas PRIVATE $<$<BOOL:${onnxruntime_USE_KLEIDIAI}>:USE_KLEIDIAI>)
+
+  if (onnxruntime_USE_KLEIDIAI)
+    # Disable the KleidiAI tests
+    set(KLEIDIAI_BUILD_TESTS  OFF)
+
+    # Fetch KleidiAI sources:
+    if (NOT TARGET kleidiai)
+      if (POLICY CMP0135)
+        cmake_policy(SET CMP0135 NEW)
+      endif()
+
+      FetchContent_Declare(kleidiai URL ${DEP_URL_kleidiai} URL_HASH SHA1=${DEP_SHA1_kleidiai})
```

Comment:
please use onnxruntime_fetchcontent_declare 
And, add EXCLUDE_FROM_ALL keyword there.